libphal/phal_pdbg.C

```diff
@@ -280,4 +280,31 @@ bool isSbeVitalAttnActive(struct pdbg_target *proc)
 	return validAttn;
 }
 
+bool hasControlTransitionedToPHYP()
+{
+  // If unable to read the register, by default consider the control is in hostboot
+  bool transitionedToPHYP = false;
+  //Read the scratch register to find if the host is transitioned to phyp
+  pdbg_target* pibTarget = pdbg_target_from_path(nullptr, "/proc0/pib");
+
+  uint64_t l_coreScratchRegData = 0xFFFFFFFFFFFFFFFFull;
+  // HB changes the below core scratch reg as one of the last instructions that is run,
+  // so if that is zero then we're in phyp
+  uint64_t l_coreScratchRegAddr = 0x4602F489;
+
+  // Is there any error in reading the scom address, assign default value to data
+  if (pib_read(pibTarget, l_coreScratchRegAddr, &l_coreScratchRegData))
+  {
+      log(level::ERROR, "scom read error: 0x%X", l_coreScratchRegAddr );
+      l_coreScratchRegData = 0xFFFFFFFFFFFFFFFFull;
+  }
+
+  // If the register reads zero, return control moved to phyp.
+  if(l_coreScratchRegData == 0)
```

Comment:
`if (0 == l_coreScratchRegData)`
    `return true;`
`return false;`
Actually the method's main aim to make sure whether the control has been successfully transferred to PHYP or not. So any false/negative result determining so should have been translated into a false interpretation of the assumption